sui_programmability/examples/frenemies/sources/frenemies.move

```diff
@@ -96,6 +103,18 @@ module frenemies::frenemies {
         scorecard.assignment = assignment::get(state, ctx);
     }
 
+    /// To prove participation offchain, participants need to hash some information and
+    /// call this function to prove that they own the address in the leaderboard.
+    ///
+    /// Aborts if the value submitted is not a hash (based on the length - 32 bytes).
+    /// Dummy check to prevent unnecessary spam and mistakes on the caller end.
+    public entry fun prove_participation(commitment_hash: vector<u8>) {
+        assert!(std::vector::length(&commitment_hash) == 32, EHashExpected);
+        event::emit(ParticipationProofCommitted {
```

Comment:
Should we include the sender address in the event to make it easier to check this based on the event stream alone?